### PR TITLE
Issue #11528: Split pitest coding into 3 to improve execution time

### LINKED ISSUE #11528: Split pitest-coding execution into two executions and investigate instability
pitest-coding is running too long in comparison to other and fails too often

we need stabilize it and make it quicker to execute.

I am open to any solutions.

related to https://github.com/checkstyle/checkstyle/issues/11427

### PR DESCRIPTION
**Pull request by Vyom-Yadav**
Resolves #11528 

No Checks are left for migration.
One check was left in the initial commit, semaphore failed - https://checkstyle.semaphoreci.com/jobs/7b20f037-132b-4ddd-ab3b-089fa7a1c702

Bash script mentioned below governs it.
https://github.com/checkstyle/checkstyle/blob/cffefba904f8d47f7bf9a9d9b053f5d33ea2afff/.ci/validation.sh#L39

---

`pitest-coding-1` -> `6 minutes`
`pitest-coding-2` -> `4 minutes`
`pitest-require-this-check` -> `18 minutes`

There is a separate instance for `RequireThis` as it consumes a lot of time. 

---

ATTENTION: Update https://github.com/checkstyle/checkstyle/wiki/How-to-run-certain-phases-and-validations#how-to-generate-pitest-report after merging this PR.

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -2499,16 +2499,205 @@
                 <mutator>VOID_METHOD_CALLS</mutator>
               </mutators>
               <targetClasses>
-                <param>com.puppycrawl.tools.checkstyle.checks.coding.*</param>
```
[the remark this answers is not in the document]
Comment (Vyom-Yadav) [reply pom.xml:2502]: How about a custom groovy or bash script?

[the remark this answers is not in the document]
Comment (rnveach) [reply pom.xml:2502]: > We need to think of some way to make sure that new checks get added to pitest execution

There should already be a bash script ensuring this. Nothing new should be needed in that regards.

[the remark this answers is not in the document]
Comment (rnveach) [reply pom.xml:2502]: @Vyom-Yadav @nick-mancuso https://github.com/checkstyle/checkstyle/blob/master/.ci/validation.sh#L39

**Comment by nick-mancuso** [comment]
@Vyom-Yadav please see failing CI

**Comment by Vyom-Yadav** [comment]
@nick-mancuso Some suppressed mutations are killed:
```
-FinalLocalVariableCheck.java.html:<td class='covered'><pre><span  class='survived'>            if (storedVariable != null &#38;&#38; isSameVariables(storedVariable, ast)) {</span></pre></td></tr>
```
```
-HiddenFieldCheck.java.html:<td class='covered'><pre><span  class='survived'>            return instanceFields.contains(field)</span></pre></td></tr>
```
```
-RequireThisCheck.java.html:<td class='covered'><pre><span  class='survived'>            if (elistToken != null &#38;&#38; ident.getText().equals(ast.getText())) {</span></pre></td></tr>
```
```
-RequireThisCheck.java.html:<td class='covered'><pre><span  class='survived'>            if (lookForMethod &#38;&#38; containsMethod(nameToFind)</span></pre></td></tr>
```

**Comment by romani** [comment]
Try to return 3 times, if not reproduced. Let's consider them as stable killed.

**Comment by romani** [comment]
I would just remove.

**Review by romani — APPROVED** [review]
Ok to merge

File: .github/workflows/pitest.yml [hunk 2]
```diff
@@ -128,7 +128,57 @@ jobs:
...
+          mkdir staging && cp -r target/pit-reports/ staging
+      - name: Archive code coverage results
+        if: failure() || github.ref == 'refs/heads/master'
+        uses: actions/upload-artifact@v2
+        with:
+          name: pitest-coding-2-coverage-report
+          path: staging
+          retention-days: 7
+
+  pitest-require-this-check:
```
Comment (nick-mancuso) [inline .github/workflows/pitest.yml:160]: Please do `pitest-coding-require-this-check` to be consistent

**Review by nick-mancuso — CHANGES_REQUESTED** [review]
MInor item:

File: .github/workflows/pitest.yml (hunk 2 above)
Comment (Vyom-Yadav) [reply .github/workflows/pitest.yml]: Done.

**Review by nick-mancuso — APPROVED** [review]

**Pull request merged**